system/lib/wasmfs/wasmfs.cpp

```diff
@@ -224,9 +271,22 @@ __wasi_fd_t __syscall_open(long pathname, long flags, long mode) {
 #endif
     curr = directory->locked().getEntry(pathParts[i]);
 
-    // Requested entry (file or directory
+    // Requested entry (file or directory)
     if (!curr) {
-      return -(ENOENT);
+      if (i == pathParts.size() - 1 && flags & O_CREAT) {
+        // If curr is the last element and the create flag is specified
+        auto dir = directory->locked();
```

Comment:
How about a convention of `lockedDir` for these, like `lockedFile`?

Reply:
Yep, sgtm